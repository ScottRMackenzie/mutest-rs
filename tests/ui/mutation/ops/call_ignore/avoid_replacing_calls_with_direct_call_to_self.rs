//@ print-mutations
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: call_delete, call_value_default_shadow

#![allow(unused)]

struct S(usize);

impl S {
    fn new(v: usize) -> Self {
        Self(v)
    }
}

impl Default for S {
    fn default() -> Self {
        // NOTE: Test with both functions and constructors, in case we ever
        //       decide to stop mutating constructor calls.
        let _ = Self::new(1);
        Self(1)
    }
}

fn f() {
    let _ = S::default();
}

#[test]
fn test() {
    f();
}
