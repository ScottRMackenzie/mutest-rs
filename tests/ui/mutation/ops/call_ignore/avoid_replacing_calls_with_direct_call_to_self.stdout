0 mutations; 0 safe; 0 unsafe (0 tainted)
